crates/autopilot/src/run.rs

```diff
@@ -572,17 +569,42 @@ async fn shadow_mode(args: Arguments) -> ! {
     let liveness = Arc::new(Liveness::new(args.max_auction_age));
     shared::metrics::serve_metrics(liveness.clone(), args.metrics_address);
 
+    let ethrpc = ethrpc(&args.shared.node_url).await;
+    let chain = ethrpc.chain();
+    let web3 = ethrpc.web3().clone();
+    let url = ethrpc.url().clone();
+    let contracts = infra::blockchain::contracts::Addresses {
+        settlement: args.shared.settlement_contract_address,
+        weth: args.shared.native_token_address,
+    };
+    let eth = ethereum(
+        web3.clone(),
+        chain,
+        url,
+        contracts.clone(),
+        args.shared.current_block.block_stream_poll_interval,
+    )
+    .await;
+
+    let cow_amm_registry = cow_amm::Registry::new(web3.clone(), eth.current_block().clone());
```

Comment:
IMO the shadow mode should not index stuff on it's own but rather expect the cow amm to be returned in the auction from the orderbook as mentioned [here](https://github.com/cowprotocol/services/pull/2796#discussion_r1663630099).